pddlgym/core.py

```diff
@@ -417,11 +495,30 @@ def sample_transition(self, action):
 
         return state, reward, done, debug_info
 
+    def sample_transition(self, action):
+        state = self._get_successor_state(self._state, action, self.domain,
+                                          inference_mode=self._inference_mode,
+                                          raise_error_on_invalid_action=self._raise_error_on_invalid_action)
+        return self._get_new_state_info(state)
```

Comment:
I took the liberty to refactor here by using the newly added `_get_new_state_info` function to avoid code duplication.